app/contexts/WalletContext.tsx

```diff
@@ -77,31 +79,24 @@ function MnemonicUnprotectedProvider (props: WalletProviderProps<MnemonicProvide
 function MnemonicEncryptedProvider (props: WalletProviderProps<EncryptedProviderData>): JSX.Element | null {
   const { network } = useNetworkContext()
   const client = useWhaleApiClient()
-  const signingRef = useRef<EncryptedWalletInterface>({
-    provide: (prompt) => { signingRef.current.prompt = prompt },
-    resolve: () => {},
-    reject: () => {}
+  const [promptUI, setPromptUI] = useState<EncryptedWalletInterface>({
+    prompt: async () => { throw new Error('Prompt UI not ready') }
   })
 
-  const promptPassphrase = useCallback(async () => {
-    return await new Promise<string>((resolve, reject) => {
-      // TODO(ivan): implementation
-      if (signingRef.current?.prompt !== undefined) {
-        signingRef.current.resolve = resolve
-        signingRef.current.reject = reject
-        signingRef.current.prompt()
-      } // else UI not ready
-    })
-  }, [])
-
   const wallet = useMemo(() => {
-    const provider = MnemonicEncrypted.initProvider(props.data, network, promptPassphrase)
+    const provider = MnemonicEncrypted.initProvider(props.data, network, promptUI)
     return initWhaleWallet(provider, network, client)
-  }, [])
+  }, [promptUI])
+
+  const encryptedWalletInterface: EncryptedWalletUIContext = {
+    provide: (ewi: EncryptedWalletInterface) => {
+      setPromptUI(ewi)
+    }
+  }
 
   return (
     <WalletContext.Provider value={wallet}>
-      <EncryptedWalletContext.Provider value={signingRef}>
+      <EncryptedWalletContext.Provider value={encryptedWalletInterface}>
```

Comment:
```suggestion
      <EncryptedWalletContext.Provider value={encryptedWalletInterface}>
        <TransactionAuthorization />
```

Reply:
Argh this make sense but implementation is impossible. The <TransactionAuthorization /> here will mess up the playground page UI in browser